src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -546,7 +546,11 @@ private static Consumer<CodeBuilder> generateTypeSwitchSkeleton(Class<?> selecto
                 Object caseLabel = caseLabels[idx];
                 cb.labelBinding(caseTargets[idx]);
                 if (caseLabel instanceof Class<?> classLabel) {
-                    if (unconditionalExactnessCheck(selectorType, classLabel)) {
+                    if (isNotValidPair(selectorType, caseLabel)){
+                        cb.goto_(next);
+                        continue;
+                    }
+                    else if (unconditionalExactnessCheck(selectorType, classLabel)) {
```

Comment:
Can we merge this into `isNotValidPair(...) || unconditionalExactnessCheck(...)` then do nothing? The next label is already in theory immediately bound to the instruction after goto.

Reply:
The empty body means that we unconditionally return the index `case 0 -> /*no if is generated*/ return 0;`.

I confirmed that by merging, the following test would not pass. It returns erroneously 0, instead of 1.

```java
testPrimitiveType((byte) 1, byte.class,0, 1, boolean.class, byte.class);
```